File: quantum-vulnerable/broken_algorithms.py
#!/usr/bin/env python3
"""Quantum-vulnerable / broken algorithms: RSA, ECDSA, 3DES.

RSA and ECDSA are broken by Shor's algorithm on a large quantum computer; 3DES
(DESede) is a weak 64-bit-block legacy cipher. Kept for legacy/interop testing.
Requires: cryptography (pip install cryptography).
"""
import os
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.asymmetric import ec, padding, rsa
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes


def rsa_oaep(data: bytes) -> bytes:
    key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
    pad = padding.OAEP(mgf=padding.MGF1(hashes.SHA256()), algorithm=hashes.SHA256(), label=None)
    return key.decrypt(key.public_key().encrypt(data, pad), pad)


def ecdsa_sign(message: bytes) -> bytes:
    key = ec.generate_private_key(ec.SECP256R1())
    signature = key.sign(message, ec.ECDSA(hashes.SHA256()))
    key.public_key().verify(signature, message, ec.ECDSA(hashes.SHA256()))
    return signature


def triple_des(block: bytes) -> bytes:
    key = os.urandom(24)                       # 3DES / DESede, 192-bit (deprecated)
    iv = os.urandom(8)
    encryptor = Cipher(algorithms.TripleDES(key), modes.CBC(iv)).encryptor()
    return encryptor.update(block) + encryptor.finalize()


if __name__ == "__main__":
    print("RSA-2048 OAEP round-trip:", rsa_oaep(b"secret"))
    print("ECDSA P-256 signature len:", len(ecdsa_sign(b"message")))
    print("3DES-CBC ciphertext len:", len(triple_des(b"8bytes!!")))

File: quantum-vulnerable/brokenAlgorithms.js
// Quantum-vulnerable / broken algorithms: RSA (Shor), ECDSA (Shor), 3DES (weak).
'use strict';
const crypto = require('crypto');

function rsaOaep(data) {
  const { publicKey } = crypto.generateKeyPairSync('rsa', { modulusLength: 2048 });
  return crypto.publicEncrypt(
    { key: publicKey, oaepHash: 'sha256', padding: crypto.constants.RSA_PKCS1_OAEP_PADDING },
    data,
  );
}

function ecdsaSign(message) {
  const { privateKey } = crypto.generateKeyPairSync('ec', { namedCurve: 'P-256' });
  return crypto.sign('sha256', message, privateKey);
}

function tripleDes(data) {
  const key = crypto.randomBytes(24);              // 3DES / des-ede3, deprecated
  const iv = crypto.randomBytes(8);
  const cipher = crypto.createCipheriv('des-ede3-cbc', key, iv);
  return Buffer.concat([cipher.update(data), cipher.final()]);
}

console.log('RSA-OAEP ciphertext:', rsaOaep(Buffer.from('secret')).length);
console.log('ECDSA signature:', ecdsaSign(Buffer.from('message')).length);
console.log('3DES ciphertext:', tripleDes(Buffer.from('12345678')).length);
module.exports = { rsaOaep, ecdsaSign, tripleDes };
